specs/waku/v2/waku-relay.md

```diff
@@ -83,13 +114,30 @@ The `subscribe` field MUST contain a boolean, where 1 means subscribe and 0 mean
 
 The `topicid` field MUST contain the topic.
 
+
+## Security Analysis 
+<!-- TODO: realized that the prime security objective of the `relay` protocol is to provide peers anonymity as such this feature is prioritized over other features e.g., anonymity is preferred over authenticity and integrity. It might be good to motivate anonymity and its impact on the relay protocol or other protocols invoking relay protocol.-->
+
+- **Message Publisher Anonymity**: To preserve message publisher anonymity, one should remove any PII from the published message. As such, `relay` protocol follows the `StrictNoSign` policy as described in [libp2p PubSub specs](https://github.com/libp2p/specs/tree/master/pubsub#message-signing). As the result of the `StrictNoSign` policy, `Message`s should be built without the  `from`, `signature` and `key` fields since each of these three fields individually counts as PII for the author of the message (one can link the creation of the message with libp2p peerId and thus indirectly with the IP address of the publisher). 
+Note that removing identifiable information from messages cannot lead to perfect anonymity. The direct connections of a publisher might be able to figure out which `Message`s belong to that publisher by analyzing its traffic. The possibility of such inference may get higher when the `data` field is also not encrypted. <!-- TODO: more investigation on traffic analysis attacks and their success probability-->
+
+## Future work
+
+- **Confidentiality**: The current version of the `relay` protocol does not address confidentiality, however, it is achievable by performing encryption on top of the `data` field of `Message`s. More details on enabling encryption mode are provided in [Libp2p PubSub specs](https://github.com/libp2p/specs/tree/master/pubsub#the-topic-descriptor) as part of the `TopicDescriptor` messages.
```

Comment:
Is this in scope of the relay protocol, or is this more something that we want to defer to Waku Message spec? So more of an informational note perhaps? Ditto below

Reply:
I agree that confidentiality of the `data` can be addressed at different protocol layers and in this case the `data` field which is a waku message could be encrypted before being passed to the `relay` protocol, however, since the `relay` protocol carries the `data` field, the decision of whether the `data`  should be encrypted/sealed or not could be (and is good to be) part of this protocol as well  (moreover, the encryption option is already offered by libp2p using `TopicDescriptor`s) 
Of course, if no encryption takes place at the `relay` level, one can always encrypt the waku message before handing to the `relay` protocol.